src/custom/pages/Claim/InvestmentFlow/InvestOption.tsx

```diff
@@ -13,78 +12,94 @@ import { ApprovalState } from 'hooks/useApproveCallback'
 
-function _scaleValue(maxValue: CurrencyAmount<Currency>, value: string) {
-  // parse percent to string, example 25% -> 4 or 50% -> 2
-  const parsedValue = new Fraction(value, '100')
-
-  // divide maxValue with parsed value to get invest amount
-  return maxValue.multiply(parsedValue).asFraction
+enum ErrorMsgs {
+  Balance = 'Insufficient balance to invest. Adjust the amount or go back to remove this investment option',
+  Input = 'Input amount is bigger then available investment amount',
 }
 
 export default function InvestOption({ approveData, claim, optionIndex }: InvestOptionProps) {
   const { currencyAmount, price, cost: maxCost } = claim
   const { updateInvestAmount } = useClaimDispatchers()
-  const { investFlowData } = useClaimState()
+  const { investFlowData, activeClaimAccount } = useClaimState()
 
-  const investedAmount = useMemo(() => investFlowData[optionIndex].investedAmount, [investFlowData, optionIndex])
+  const { account } = useActiveWeb3React()
 
   const [percentage, setPercentage] = useState<string>(INVESTMENT_STEPS[0])
+  const [typedValue, setTypedValue] = useState<string>('0')
+  const [inputError, setInputError] = useState<string>('')
 
-  const { account } = useActiveWeb3React()
+  const investedAmount = investFlowData[optionIndex].investedAmount
 
   const token = currencyAmount?.currency
-
   const balance = useCurrencyBalance(account || undefined, token)
 
   const decimals = balance?.currency?.decimals
 
-  const handleStepChange = useCallback(
+  const isSelfClaiming = account === activeClaimAccount
+
+  // on invest max amount click handler
+  const setMaxAmount = useCallback(() => {
+    if (!maxCost || !balance) {
+      return
+    }
+
+    const value = maxCost.greaterThan(balance) ? balance : maxCost
+    const amount = value.quotient.toString()
+
+    updateInvestAmount({ index: optionIndex, amount })
+    setTypedValue(formatSmart(value, decimals) || '')
+    setPercentage(INVESTMENT_STEPS[INVESTMENT_STEPS.length - 1])
+  }, [balance, decimals, maxCost, optionIndex, updateInvestAmount])
+
+  // on input field change handler
+  const onInputChange = useCallback(
     (value: string) => {
-      if (!maxCost || !balance) {
+      setTypedValue(value)
+      setInputError('')
+
+      const input = parseFloat(value)
+
+      if (!value.length || isNaN(input)) {
+        updateInvestAmount({ index: optionIndex, amount: '0' })
+        setPercentage('0')
         return
       }
 
-      const scaledCurrencyAmount = _scaleValue(maxCost, value)
+      if (!value || !maxCost || !balance || !token) {
+        return
+      }
 
-      updateInvestAmount({ index: optionIndex, amount: scaledCurrencyAmount.quotient.toString() })
-      setPercentage(value)
-    },
-    [balance, maxCost, optionIndex, updateInvestAmount]
-  )
+      // parse input to BitInt string
+      const parsed = CurrencyAmount.fromRawAmount(token, parseUnits(input.toString(), decimals).toString())
```

Comment:
Use the string `value`, don't use the float representation of that

Floats are bad!